src/index.js

```diff
@@ -0,0 +1,273 @@
+
+  return Object.entries(jobCounts)
+    .filter(function ([_, count]) {
+      return count === maxCount;
+    })
+    .map(function ([job]) {
+      return job;
+    });
+}
+
+console.log(getMostCommonJobs());
+
+// 3
+function getCompaniesWithEmployees() {
+  const employeesByCompany = employees.reduce((accumulator, employee) => {
+    if (!accumulator[employee.companyId]) {
+      accumulator[employee.companyId] = [];
+    }
+    accumulator[employee.companyId].push(employee);
+    return accumulator;
+  }, {});
+
+  return Object.entries(employeesByCompany).reduce(
+    (accumulator, [companyId, companyEmployees]) => {
+      const company = companiesMap[companyId];
+      if (company) {
+        accumulator[company.name] = companyEmployees;
+      }
+      return accumulator;
+    },
+    {},
+  );
+}
+
+console.log(getCompaniesWithEmployees());
+
+// 4
+function getTopManager(employeeId) {
+  const employeesMap = employees.reduce((accumulator, employee) => {
+    accumulator[employee.id] = employee;
+    return accumulator;
+  }, {});
+
+  let currentEmployee = employeesMap[employeeId];
+
+  if (!currentEmployee) {
+    return null;
+  }
+
+  while (currentEmployee.managerId !== null) {
+    currentEmployee = employeesMap[currentEmployee.managerId];
+  }
+
+  return currentEmployee;
+}
+
+console.log(getTopManager(32));
+
+//5
+function getDirectEmployeesByManager(managerId) {
+  return employees.filter((employee) => employee.managerId === managerId);
+}
+
+console.log(getDirectEmployeesByManager(10));
+
+//6
+function getEmployeesGroupedByManagers(employeesList) {
+  const employeesMap = employeesList.reduce((accumulator, employee) => {
+    accumulator[employee.id] = { ...employee, employees: [] };
+    return accumulator;
+  }, {});
+
+  employeesList.forEach((employee) => {
+    if (employee.managerId !== null) {
+      const manager = employeesMap[employee.managerId];
+      if (manager) {
+        manager.employees.push(employeesMap[employee.id]);
+      }
+    }
+  });
+
+  return employeesList
+    .filter((employee) => employee.managerId === null)
+    .map((manager) => employeesMap[manager.id]);
+}
+
+console.log(getEmployeesGroupedByManagers(employees));
+
+// 7
+const products = [
+  'Wireless Bluetooth Headphones',
+  'Bluetooth Wireless Earbuds',
+  'Noise Cancelling Bluetooth Headphones',
+  'Gaming Mouse with RGB Lights',
+  'RGB Gaming Mouse with Programmable Buttons',
+  'Bluetooth Noise Cancelling Earbuds',
+  'Wireless Noise Cancelling Headphones',
+  'Ergonomic RGB Gaming Mouse with Extra Buttons',
+  'Programmable RGB Wireless Gaming Mouse with Adjustable DPI',
+  'High-Performance Wireless Bluetooth Noise Cancelling Headphones',
+];
+
+// get normalized words from a product title
+const getWords = (product) => {
+  return product.toLowerCase().split(' ');
+};
+
+// function to calculate similarity between two products
+const calculateSimilarity = (product1, product2) => {
```

Comment:
Avoid using numbers in variable names for the sake of readability. Try `productOne`, `productTwo`, `productOneWords` etc.